### PR TITLE
CLI: Add webpack5 builder to CRA5 `sb init`

### PR DESCRIPTION
**Pull request by shilman**
Issue: #16150

## What I did

- [x] Install webpack5 builder on fresh CRA projects
- [x] Version `@storybook/preset-create-react-app` based on CRA version?

## How to test

```sh
yarn create react-app --scripts-version=@next --template=cra-template@next cra-sb-fresh
cd cra-sb-fresh
/path/to/storybook/lib/cli/bin/index.js init
```

### CODE REVIEW

File: lib/cli/src/generators/REACT_SCRIPTS/index.ts [hunk 1]
```diff
@@ -20,10 +22,18 @@ const generator: Generator = async (packageManager, npmOptions, options) => {
       }
     : {};
 
-  await baseGenerator(packageManager, npmOptions, options, 'react', {
+  const craVersion = semver.coerce(
+    packageManager.retrievePackageJson().dependencies['react-scripts']
+  )?.version;
+  const isWebpack5 = semver.gte(craVersion, '5.0.0');
```
Comment (kylegach) [inline lib/cli/src/generators/REACT_SCRIPTS/index.ts:28]:
```suggestion
  const isCra5 = semver.gte(craVersion, '5.0.0');
```

_Super_ minor, but the check is for the CRA version, not webpack.

**Review by kylegach — APPROVED** [review]
Looks (and works) great!

File: lib/cli/src/generators/REACT_SCRIPTS/index.ts (hunk 1 above)
Comment (mrmckeb) [reply lib/cli/src/generators/REACT_SCRIPTS/index.ts:28]: I'd suggest always coercing a version first, as this catches `-beta.0`, etc.
```ts
semver.gte(semver.coerce('42.6.7.9.3-alpha'))
```

Comment (mrmckeb) [reply lib/cli/src/generators/REACT_SCRIPTS/index.ts:28]: I don't mind the `isWebpack5` var. `isCra5` seems more logical - but then you might need to add a comment like "CRAv5 ships with WP5".

**Pull request merged**